### PR TITLE
Feature/jenkins2

### PR DESCRIPTION
**Pull request by polyakrecords**
[VSTS](https://msmobilecenter.visualstudio.com/Mobile-Center/Krukow-M1-Team/_workitems/edit/40034)
- added Jenkinsfile for Jenkins 2.0 CI
- added workaround for rspec tests

### CODE REVIEW

**Comment by polyakrecords** [comment]
@jmoody Do we plan to update macOS on CI machine?

I've added a workaround here – https://github.com/calabash/run_loop/pull/688/files#diff-46d5f187594470d0410fd8f22a068216R1314

File: spec/lib/core_simulator_spec.rb [hunk 1]
```diff
@@ -1311,6 +1311,12 @@ def create_sandbox_dirs(sub_dir_names)
 
       Resources.shared.alt_xcode_install_paths.each do |path|
         it "#{path}" do
+          # On jenkins machine we have macOS 10.12.6 installed
+          # Xcode >= 9.3 requires at least macOS 10.13
+          # so we skip these tests for now
+          if !ENV["JENKINS_HOME"].nil? && !path[/9\.[3-9]/].nil?
+            skip "Can't test against Xcode >= 9.3 on CI machine with macOS 10.12.6"
+          end
```
Comment (jmoody) [inline spec/lib/core_simulator_spec.rb]: I installed Xcode 9.3 on the macOS machine by mistake.  We should delete the Xcode 9.3 install.

Sometime this summer we will update the Jenkins machine.

Comment (jmoody) [reply spec/lib/core_simulator_spec.rb]: Xcode 9.3 has been removed from the CI machine.

We can remove this check.

File: scripts/ci/jenkins/simctl.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+# Ensure the correct CoreSimulatorService is loaded.
+
+xcrun simctl help >/dev/null 2>&1
+xcrun simctl help >/dev/null 2>&1
+xcrun simctl help >/dev/null 2>&1
```
Comment (jmoody) [inline scripts/ci/jenkins/simctl.sh]: Replace with this:

https://github.com/calabash/DeviceAgent.iOS/blob/develop/bin/simctl.sh

File: scripts/ci/jenkins/simctl.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	for try in {1..4}; do
+    local valid=$(valid_core_sim_service)
+    if [ "${valid}" = "false" ]; then
+      info "Trying again to ensure valid CoreSimulatorService"
+      sleep 1.0
+    else
+      info "CoreSimulatorService is valid"
+      break
+    fi
+	done
```
Comment (jmoody) [inline scripts/ci/jenkins/simctl.sh]: Indent looks wrong.

**Pull request merged**